exercises/perfect-numbers/example.js

```diff
@@ -1,64 +1,49 @@
-
-export default class PerfectNumbers {
-
-  /**
-   * Calculate all the divisors for a given number.
-   * @param {number} number - natural number.
-   * @returns {array} array that contains the divisors for a given number NOT including the number itself.
-   */
-  getDivisors(number) {
-    let i;
-    const divs = new Array();
-
-    // Accepts only natural numbers greater than 1.
-    if (number <= 1) {
-      return divs;
-    }
-
-    // 1 always divides everyone!
-    divs.push(1);
-
-    // Calculate the divisors up the the half of the number + 1
-    for (i = 2; i <= number / 2; i++) {
-      if (number % i === 0) {
-        divs.push(i);
-      }
-    }
-
+/**
+  * Calculate all the divisors for a given number.
+  * @param {number} number - natural number.
+  * @returns {array} that contains the divisors for a given number excluding the number itself.
+  */
+function getDivisors(number) {
+  const divs = [];
+
+  // Accepts only natural numbers greater than 1.
+  if (number <= 1) {
     return divs;
   }
 
-  /**
-   * Classify a given number as perfect, abundant or deficient.
-   * @param {number} number - number to be classified. Note: if the number is equal or smaller than 0,
-   * then returns an error: Classification is only possible for natural numbers.
-   * @returns {string} - string that contains the number classification (perfect, abundant or deficient).
-   */
-  classify(number) {
-    let sum,
-      result;
+  // 1 always divides everyone!
+  divs.push(1);
 
-    // Check if the input is valid
-    if (number <= 0) {
-      throw new Error('Classification is only possible for natural numbers.');
+  // Calculate the divisors up the the half of the number + 1
+  for (let i = 2; i <= number / 2; i += 1) {
+    if (number % i === 0) {
+      divs.push(i);
     }
+  }
 
-    // Factorize the current number.
-    const divsArray = this.getDivisors(number);
+  return divs;
+}
 
-    // Sum the factors.
-    sum = divsArray.reduce((sum, div) => sum += div, 0);
+/**
+  * Classify a given number as perfect, abundant or deficient.
+  * @param {number} - number to be classified. Note: if the number is equal or smaller than 0,
+  * then returns an error: Classification is only possible for natural numbers.
+  * @returns {string} - string that contains the classification (perfect, abundant or deficient).
+  */
+export const classify = (number) => {
+  // Check if the input is valid
+  if (number <= 0) {
+    throw new Error('Classification is only possible for natural numbers.');
+  }
 
-    // Check if the number is perfect.
-    if (sum === number) {
-      result = 'perfect';
-    } else if (sum > number) {
-      result = 'abundant';
-    } else {
-      result = 'deficient';
-    }
+  // Factorize the current number.
+  const divsArray = getDivisors(number);
 
-    return result;
-  }
+  // Sum the factors.
+  const sum = divsArray.reduce((partialSum, div) => partialSum + div, 0);
```

Comment:
One small thing here:  `Array.reduce` will use the value of first element of the list as the initial accumulator value when you don't supply an initial value explicitly.  

In effect, instead of adding zero to the first element in the first pass, you start by adding the first element to the second one, and continue from there:

```suggestion
const sum = divsArray.reduce((partialSum, div) => partialSum + div);
```

Reply:
Thanks for your suggestion! Unfortunately, it is possible for `divsArray` to be empty (consider edge case when classifying number `1`).  When `divsArray` is empty, the initial accumulator `0` is needed in order to arrive at the correct aliquot sum.  Without the initial accumulator, the `reduce` would raise an error on an empty array.

Reply:
Good point.  I did not consider that. 😁